roles/redis/tasks/main.yaml

```diff
@@ -1,14 +1,39 @@
 ---
-- name: Install redis
-  ansible.builtin.package:
-    name: "redis"
+- name: Pull Redis container image
+  containers.podman.podman_image:
+    name: "{{ redis_image }}"
     state: present
 
+- name: Create directory for Redis data
+  ansible.builtin.file:
+    path: /var/lib/redis
+    state: directory
+    owner: 1001
+    group: 1001
+    mode: '0755'
+
+- name: Run Redis as a container
+  containers.podman.podman_container:
+    name: redis
+    image: "{{ redis_image }}"
+    state: quadlet
+    volumes:
+      - /var/lib/redis:/data:Z
+    ports:
+      - "6379:6379"
+    quadlet_options:
+      - |
+        [Install]
+        WantedBy=default.target
+
+- name: Run daemon reload
+  ansible.builtin.systemd:
+    daemon_reload: true
+
 - name: Start the Redis Service
   ansible.builtin.systemd:
     name: redis
-    enabled: true
```

Comment:
Any reason `enabled` is no longer needed? Normally that's still needed if you want it to start on reboot.